sql/catalyst/src/main/antlr3/org/apache/spark/sql/catalyst/parser/FromClauseParser.g

```diff
@@ -151,8 +151,8 @@ fromSource
 @after { gParent.popMsg(state); }
     :
     (LPAREN KW_VALUES) => fromSource0
-    | (LPAREN) => LPAREN joinSource RPAREN -> joinSource
     | fromSource0
+    | (LPAREN joinSource) => LPAREN joinSource RPAREN -> joinSource
```

Comment:
Yes, it matters. Without changing the order, `select * from ((select 1 as a) union (select 1 as a)) t` doesn't work because `((select 1 as a) union` will be parsed as `LPAREN joinSource RPAREN` first and the parser will complain wrong input '(' expecting ) near union.